pkgs/test_reflective_loader/lib/test_reflective_loader.dart

```diff
@@ -154,12 +154,20 @@ void _addTestsIfTopLevelSuite() {
         if (allGroups || group.isSolo) {
           for (var test in group.tests) {
             if (allTests || test.isSolo) {
-              test_package.test(test.name, test.function,
+              test_package.test(test.name, () async {
+                await group.ensureSetUpClass();
+                test.function();
+              },
                   timeout: test.timeout,
                   skip: test.isSkipped,
                   location: test.location);
             }
           }
+          // only "tearDownClass" if class set-up was ensured ie at least one
+          // test of the group ran
+          if (group.didEnsureSetupClass) {
```

Comment:
Maybe check it inside `group.tearDownClass()`?
We probably don't need to worry that we run no-op `test_package.tearDownAll`.
Actually as is it probably will not work, because at this point we did not run any test yet, we just tell the test package what to run.